src/app/pages/common/entity/utils.ts

```diff
@@ -265,4 +265,183 @@ export class EntityUtils {
 
     return result;
   }
+
+  createRelations(relations, parentName) {
+    const result = relations.map(relation => {
+      let relationFieldName = relation[0];
+      if (parentName) {
+        relationFieldName = `${parentName}${FORM_KEY_SEPERATOR}${relationFieldName}`;
+      }
+  
+      return {
+        action: 'SHOW',
+        when: [{
+          name: relationFieldName,
+          operator: relation[1],
+          value: relation[2],
+        }]
+      };
+    });
+
+    return result;    
+  }
+
+  parseSchemaFieldConfig(schemaConfig, parentName=null, parentIsList=false) {
```

Comment:
We should have classes and types where feasible for cleaner implementation and maintenance

Reply:
Regarding using class, they are all dynamic so we can't know its structure beforehand.
Also it is related with dynamic chart release and all data is from schema that middleware sends and it has unexpected structure.
That's why we can only use `any `or `object` here.